core/templates/pages/skill-editor-page/services/skill-editor-state.service.ts

```diff
@@ -17,206 +17,223 @@
  * in the skill editor.
  */
 
-require('domain/editor/undo_redo/undo-redo.service.ts');
-require('domain/question/question-backend-api.service.ts');
-require('domain/skill/SkillObjectFactory.ts');
-require('domain/skill/skill-backend-api.service.ts');
-require('domain/skill/skill-rights-backend-api.service.ts');
-require('pages/skill-editor-page/skill-editor-page.constants.ajs.ts');
-require('services/alerts.service.ts');
-require('services/questions-list.service.ts');
-import { EventEmitter } from '@angular/core';
+import cloneDeep from 'lodash/cloneDeep';
+
+import { EventEmitter, Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
+
+import { UndoRedoService } from 'domain/editor/undo_redo/undo-redo.service';
+import { SkillBackendApiService } from 'domain/skill/skill-backend-api.service';
 import { SkillRights } from 'domain/skill/skill-rights.model';
+import { SkillRightsBackendApiService } from 'domain/skill/skill-rights-backend-api.service';
+import { Skill, SkillObjectFactory } from 'domain/skill/SkillObjectFactory';
+import { AlertsService } from 'services/alerts.service';
+import { QuestionsListService } from 'services/questions-list.service';
 
-angular.module('oppia').factory('SkillEditorStateService', [
-  '$rootScope', 'AlertsService', 'QuestionsListService',
-  'SkillBackendApiService', 'SkillObjectFactory',
-  'SkillRightsBackendApiService', 'UndoRedoService',
-  function(
-      $rootScope, AlertsService, QuestionsListService,
-      SkillBackendApiService, SkillObjectFactory,
-      SkillRightsBackendApiService, UndoRedoService) {
-    var _skill = SkillObjectFactory.createInterstitialSkill();
-    var _skillRights = (
-      SkillRights.createInterstitialSkillRights());
-    var _skillIsInitialized = false;
-    var assignedSkillTopicData = null;
-    var _skillIsBeingLoaded = false;
-    var _skillIsBeingSaved = false;
-    var _groupedSkillSummaries = {
-      current: [],
-      others: []
-    };
-    var _skillChangedEventEmitter = new EventEmitter();
-    var _setSkill = function(skill) {
-      _skill.copyFromSkill(skill);
-      _skillChangedEventEmitter.emit();
-      _skillIsInitialized = true;
-    };
-
-    var _updateSkill = function(skill) {
-      _setSkill(skill);
-    };
-
-    var _updateGroupedSkillSummaries = function(groupedSkillSummaries) {
-      var topicName = null;
-      _groupedSkillSummaries.current = [];
-      _groupedSkillSummaries.others = [];
-
-      for (var name in groupedSkillSummaries) {
-        var skillSummaries = groupedSkillSummaries[name];
-        for (var idx in skillSummaries) {
-          if (skillSummaries[idx].id === _skill.getId()) {
-            topicName = name;
-            break;
-          }
-        }
-        if (topicName !== null) {
+interface gsummaries {
+  current: {
+    id: string,
+    description: string,
+  }[],
+  others: unknown[]
```

Comment:
Would this be of type `SkillSummaryBackendDict[]` (looking at the code flow for _updateGroupedSkillSummaries seems to indicate this)?

Reply:
Seems SkillSummaryBackendDict  is correct